### ISSUE TITLE
Reward stays at 0 and eventually loss becomes 0 too

### ISSUE THREAD

**Issue by bhavitvyamalik** [idanshen/Self-Distillation#14]
I'm trying to run this script to train Qwen2.5-7B-Instruct for tool-use. I followed exact same steps to install libraries. My arguments for training:
```
python main.py \
  --dataset_name tooluse \
  --model_name $MODEL_PATH \
  --output_dir $OUTPUT_PATH \
  --ref_model_mixup_alpha 1.0 \
  --learning_rate 5e-5 \
  --num_train_epochs 2 
```
This is how my train loss curve looks like:

<img width="682" height="604" alt="Image" src="https://github.com/user-attachments/assets/edd700e9-1780-46e6-b06b-882ec2fea747" />

Furthermore, `train/rewards` is always 0 in the training. I looked at the generations and model is somehow not able to predict the correct function but loss keeps on going down. The moment train loss becomes 0, this is what model generates for any prompt:
`car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car c`

I even tried Qwen3-1.7B but got same results.

**Comment by liboqiao1234** [comment]
So have you solved this problem? I am under the same situation when I tried to adapt to a new dataset
